updates versions callout (#8313)
fishtown-analytics/docs.getdbt.com: website/snippets/_model-version-callout.md

:::info Model versions, dbt_project.yml versions, and .yml versions

The word "version" appears in multiple places in docs site and with different meanings:

> - [Model versions](/docs/mesh/govern/model-versions) - A dbt <Constant name="mesh" /> feature that enables better governance and data model management by allowing you to track changes and updates to models over time

Reviewer: ```suggestion
- [Model versions](/docs/mesh/govern/model-versions) &mdash; A dbt <Constant name="mesh" /> feature that enables better governance and data model management by allowing you to track changes and updates to models over time.
```

> - [dbt_project.yml version](/reference/project-configs/version#dbt_projectyml-versions) - `dbt_project.yml` version is unrelated to <Constant name="mesh" /> and refers to the compatibility of the dbt project with a specific version of dbt. (Optional since dbt v1.5)

Reviewer: Since 1.5 has been EOL for a bit, we can probably remove the disclaimer and just mark it as optional
```suggestion
- [dbt_project.yml version](/reference/project-configs/version#dbt_projectyml-versions)(optional) &mdash; `dbt_project.yml` version is unrelated to <Constant name="mesh" /> and refers to the compatibility of the dbt project with a specific version of dbt.
```

> - [.yml property file version](/reference/project-configs/version#yml-property-file-versions) - Version numbers within .yml property files inform how dbt parses those YAML files. Unrelated to <Constant name="mesh" />. (Optional since dbt v1.5) - 

Reviewer: ```suggestion
- [.yml property file version](/reference/project-configs/version#yml-property-file-versions)(optional) &mdash; Version numbers within .yml property files inform how dbt parses those YAML files. Unrelated to <Constant name="mesh" />.
```
